ahnlich/ai/src/engine/store.rs

```diff
@@ -185,20 +220,95 @@ impl AIStoreHandler {
         self.stores.clear(&guard);
         store_length
     }
+
+    #[tracing::instrument(skip(self))]
+    pub(crate) fn process_store_input(
+        &self,
+        process_action: &PreprocessAction,
+        input: &mut StoreInput,
+        index_model: &AIModel,
+    ) -> Result<(), AIProxyError> {
+        match process_action {
```

Comment:
I think a match over (process_action, store_input) would be cleaner and you can error in the right places e.g

```rust
match (process_action, store_input) {
    (PreprocessAction::Image(_), StoreInput::RawString(_)) => Err("Used Image preprocessing for String type"),
    ...
}
```